### ISSUE TITLE
Add block to prevent editing of studies shared with curation app?

### ISSUE THREAD

**Issue by jimallman** [OpenTreeOfLife/phylografter#126]
@pmidford, we're talking about whether it's possible to move the block on editing "shared" studies (those available to both phylografter and the new curation app) to block them on phylografter. It seems like we could do this by querying oti to see if the current study's ID is already there, yes?

The reason we're interested in this is that there are some "downstream" edits (prep for synthesis) that can't currently be done in phylografter, like:
- re-rooting (and marking of roots as meaningful)
- marking one or more trees as "preferred"

**Comment by mtholder** [comment]
since we were talking about adding new studies in the new tool, and frequently importing to the new curator from pg, would this block all edits in phylografter? Would there be any IDs not in oti?

**Comment by pmidford** [comment]
I don't think it would be difficult to implement such a block; I agree with Mark that this might block all edits in Phylografter, unless there was also a block so that some studies in phylografter would not get picked up by oti (so someone could start editing a new study in phylografter and release it when they were ready to use the curator tool or to offer it for synthesis).

**Comment by jimallman** [comment]
Hm, yes. Maybe an explicit "hand-off" from one to the other? Argh, it's all half measures.

How far are we from robust sharing between both tools?

**Comment by pmidford** [comment]
On 6/9/14, 5:49 PM, Jim Allman wrote:

> Hm, yes. Maybe an explicit "hand-off" from one to the other? Argh, 
> it's all half measures.
> 
> How far are we from robust sharing between both tools?
> 
> Good question. I'd like to think weeks, but I've been thinking that for 
> a while now.   Phylografter ingest from phylesystem seems to be working, 
> but still rather slow.  There are a couple of changes in phylografter 
> #121 that I will be finishing up early tomorrow.  There are a couple of 
> things Mark identified that I need to add to phylografter (mostly 
> capturing SHA's from ingested studies).  Then I can set up the 'chron' 
> job on the phylografter side and we can try an update from a dev 
> phylesystem that has studies that were written by the curator tool and 
> round trip to make sure everything the curator tool expects to see is in 
> the right place.

Error handling is also important, both recognizing errors (our API 
documentation doesn't have much to say about this) and what phylografter 
should do if it can't push an update back to phylesystem.

Those are the issues I'm aware of that are holding us back.

-Peter

**Comment by jimallman** [comment]
> Those are the issues I'm aware of that are holding us back.

No pressure, I was just wondering. Thanks for the update!
